server/src/main/java/org/apache/cassandra/sidecar/acl/authorization/RoleAuthorizationsCache.java

```diff
@@ -0,0 +1,97 @@
+ * are picked up during cache refreshes.
+ */
+@Singleton
+public class RoleAuthorizationsCache extends AuthCache<String, Map<String, Set<Authorization>>>
+{
+    private static final String NAME = "role_permissions_cache";
+    protected static final String UNIQUE_CACHE_ENTRY = "unique_cache_entry_key";
+
+    @Inject
+    public RoleAuthorizationsCache(Vertx vertx,
+                                   ExecutorPools executorPools,
+                                   SidecarConfiguration sidecarConfiguration,
+                                   SystemAuthDatabaseAccessor systemAuthDatabaseAccessor,
+                                   SidecarPermissionsDatabaseAccessor sidecarPermissionsDatabaseAccessor)
+    {
+        super(NAME,
+              vertx,
+              executorPools,
+              k -> loadAuthorizations(systemAuthDatabaseAccessor,
+                                      sidecarConfiguration.serviceConfiguration().schemaKeyspaceConfiguration().isEnabled(),
+                                      sidecarPermissionsDatabaseAccessor),
+              () -> Collections.singletonMap(UNIQUE_CACHE_ENTRY,
+                                             loadAuthorizations(systemAuthDatabaseAccessor,
+                                                                sidecarConfiguration.serviceConfiguration().schemaKeyspaceConfiguration().isEnabled(),
+                                                                sidecarPermissionsDatabaseAccessor)),
+              sidecarConfiguration.accessControlConfiguration().permissionCacheConfiguration());
+    }
+
+    public Set<Authorization> getAuthorizations(String role)
+    {
+        return get(UNIQUE_CACHE_ENTRY).get(role);
+    }
+
+    private static Map<String, Set<Authorization>> loadAuthorizations(SystemAuthDatabaseAccessor systemAuthDatabaseAccessor,
+                                                                      boolean isSidecarSchemaEnabled,
+                                                                      SidecarPermissionsDatabaseAccessor sidecarPermissionsDatabaseAccessor)
+    {
+
+        // when entries in cache are not found, null is returned. We can not add null in Map
+        Map<String, Set<Authorization>>  roleAuthorizations
+        = Optional.ofNullable(systemAuthDatabaseAccessor.getAllRolesAndPermissions()).orElse(Collections.emptyMap());
+
+        if (isSidecarSchemaEnabled)
+        {
+            Map<String, Set<Authorization>> sidecarAuthorizations
+            = Optional.ofNullable(sidecarPermissionsDatabaseAccessor.getAllRolesAndPermissions()).orElse(Collections.emptyMap());
+
+            // merge authorizations from Cassandra and Sidecar tables
+            sidecarAuthorizations.forEach((role, permissions) -> {
+                roleAuthorizations.merge(role, permissions, (existingPermissions, newPermissions) -> {
+                    existingPermissions.addAll(newPermissions);
+                    return existingPermissions;
+                });
```

Comment:
Can `authorizations` be used  for Set<Authorization> instead of using `permissions`?

Reply:
sure